docs/runtime-contract.md

```diff
@@ -211,9 +211,9 @@ following protocol will be used:
 - `h2c`: HTTP/2 transport, as described in
   [section 3.4 of the HTTP2 spec (Starting HTTP/2 with Prior Knowledge)](https://http2.github.io/http2-spec/#known-http)
 
-Developers SHOULD prefer to use automatic content negotiation where available,
+Developers SHOULD use automatic content negotiation where available,
 and MUST NOT set the `name` field to arbitrary values, as additional transports
```

Comment:
@dgerd is the "MUST NOT" here overstepping too? Seems like it given we're removing the other keywords from this paragraph.

Reply:
The "MUST NOT" is important to allow us to reserve the 'name' field for future usage. It could be argued that this is more of an API constraint than a runtime constraint, but it is important and I am okay with duplicating it here for clarity.

For example, if HTTP/3.0 comes along and we want to allow people to specify "http3" we can only enable this we know that no existing Knative Configurations allow "http3"; otherwise this will be a breaking change.

I am okay rephrasing this if you think it is unclear as is.

Reply:
I think the "MUST NOT" suggests that validation during admission control is allowed here, and containers with other port names may be rejected.

Reply:
ok I'll leave it as-is.